Error Handling

emanuelmetal opened the pull request:
# Background
We need to avoid throwing errors without the proper handling, instead add a logging system to log those errors, and in the future send them to a log service.

# Changes Done
- Added js-logger library to encapsulate the logging functionality on a basic configuration
- As per some error on develop, **husky** a pre-commit, pre-hook library was added.

Commit:
added global logger system

Commit:
Fix delete user and add pre commit/push hooks

Commit:
remove precommit hook, leave prepush

mravinale commented on webapp/app/actions/usersActions.js:
```diff
@@ -117,14 +113,13 @@ export function getUsers(queryParams = DEFAULT_PAGINATION_QUERY) {
+  return dispatch =>{
+    logger.error(error.message);
+    dispatch(loadingUsersFailed(error));
+  };
```
I think we would need something global for the whole application, can we do that?

emanuelmetal replied:
Yes, what I think is to set up a logging service that will log (and potentially send them to a server) and leave here the dispatch of the fail, which only purpose is to update the store with the error flag to show some feedback.
@mravinale Is that what you are asking?

Commit:
global error logging service

Commit:
Log service basic implementation

Commit:
add and fix some UT

Commit:
UT fixes, and several changes

Increased coverage to 100% of branches
Removed babel js-logger plugin
Added lodash get sinlge method
Added webpack bundle  analyzer

IvanScoles commented on webapp/app/utils/errorService.test.js:
```diff
@@ -0,0 +1,58 @@
+    });
+  });
+
+  // describe('Set Log level', () => {
```
We should remove this describe?

emanuelmetal replied:
Yes i'll do it

Commit:
improved webpack analyzer

emanuelmetal commented:
@mravinale check this new changes please

Commit:
removed commented code

mravinale commented on webapp/app/actions/usersActions.js:
```diff
@@ -117,14 +113,11 @@ export function getUsers(queryParams = DEFAULT_PAGINATION_QUERY) {
-  }
-  return response;
+function handleErrors(error, dispatch) {
+  errorService.logErrors('action failed', 'userActions.js');
```
what if we send the error object? or the message inside the object to the error service, now all the errors are llike hardcoded

mravinale replied:
proposal:
create the 3rd argument in order to pass a function to be executed.
````
errorService.logErrors(error.message 'userActions.js', ()=> somefunction());
````

having that we can remove handleErrors function and use directly the errorService

mravinale commented:
@emanuelmetal what do you think implementing this as well https://medium.com/leanjs/react-16-error-handing-with-componentdidcatch-fa4e248d45d7

Commit:
updated yarn lock and jest config
